app/controllers/posts_controller.rb

```diff
@@ -1,27 +1,19 @@
 # encoding:UTF-8
 class PostsController < ApplicationController
   load_permissions_and_authorize_resource
-  before_action :set_council
+  load_resource :council, find_by: :url
 
   def remove_profile
-    @profile = Profile.find_by_id(params[:profile_id])
-    @post.profiles.delete(@profile)
-    respond_to do |format|
-      format.html { redirect_to council_posts_path(@council), notice: @profile.name.to_s + ' har inte längre posten ' + @post.title.to_s + '.' }
-    end
+    profile = Profile.find_by_id(params[:profile_id])
+    @post.remove_profile(profile)
+    redirect_to council_posts_path(@council),
+                notice: %(#{profile.full_name} har inte längre posten #{@post.title}.)
   end
 
   def add_profile_username
     @user = User.find_by(username: params[:username])
-    if @user != nil
-      @profile = @user.profile
-    end
-    if @profile == nil
-      respond_to do |format|
-        format.html { redirect_to council_posts_path(@council), flash: {alert: 'Hittade ingen användare med det användarnamnet.'} }
-      end
-    elsif @profile.name.blank?
-      redirect_to council_posts_path(@council), flash: {alert: 'Användaren :"' + @user.username.to_s + '" måste fylla i fler uppgifter i sin profil.'}
+    if @post.add_profile(@user)
+      redirect_to council_posts_path(@council), flash: {alert: 'Hittade ingen användare med det användarnamnet.'}
```

Comment:
Line is too long. [113/100]<br>Space inside { missing.<br>Space inside } missing.